src/mpid/ch4/src/ch4_impl.h

```diff
@@ -551,6 +579,7 @@ do {                                                                  \
 #define MPIDI_CH4U_EPOCH_FENCE_CHECK(win, mpi_errno, stmt)              if (0) goto fn_fail;
 #define MPIDI_CH4U_EPOCH_POST_CHECK(win, mpi_errno, stmt)               if (0) goto fn_fail;
 #define MPIDI_CH4U_EPOCH_LOCK_CHECK(win, mpi_errno, stmt)               if (0) goto fn_fail;
+#define MPIDI_CH4U_EPOCH_PER_TARGET_LOCK_CHECK(win, rank, mpi_errno, stmt)  if (0) goto fn_fail;
```

Comment:
Did you try building without `--enable-error-checking=all`? Note that you need to provide two definitions for one macro, one for `HAVE_ERROR_CHECKING` and the other for `!HAVE_ERROR_CHECKING`